src/stp/plays/defensive/DefendShot.cpp

```diff
@@ -23,52 +23,118 @@ DefendShot::DefendShot() : Play() {
+    flagMap.insert({"waller_1", {DealerFlagPriority::HIGH_PRIORITY, {closeToOurGoalFlag}}});
+    flagMap.insert({"waller_2", {DealerFlagPriority::HIGH_PRIORITY, {closeToOurGoalFlag}}});
+    flagMap.insert({"waller_3", {DealerFlagPriority::HIGH_PRIORITY, {closeToOurGoalFlag}}});
     flagMap.insert({"defender_1", {DealerFlagPriority::MEDIUM_PRIORITY, {closeToOurGoalFlag}}});
     flagMap.insert({"defender_2", {DealerFlagPriority::MEDIUM_PRIORITY, {closeToOurGoalFlag}}});
+    flagMap.insert({"midfielder_1", {DealerFlagPriority::MEDIUM_PRIORITY, {notImportant}}});
+    flagMap.insert({"midfielder_2", {DealerFlagPriority::MEDIUM_PRIORITY, {notImportant}}});
+    flagMap.insert({"midfielder_3", {DealerFlagPriority::MEDIUM_PRIORITY, {notImportant}}});
+    flagMap.insert({"midfielder_4", {DealerFlagPriority::MEDIUM_PRIORITY, {notImportant}}});
 
     return flagMap;
 }
 
 void DefendShot::calculateInfoForRoles() noexcept {
+    calculateInfoForWallers();
     calculateInfoForDefenders();
     calculateInfoForHarassers();
     calculateInfoForKeeper();
 }
 
+void DefendShot::calculateInfoForWallers() noexcept{
+    stpInfos["waller_1"].setPositionToDefend(field.getOurTopGoalSide());
+    stpInfos["waller_1"].setBlockDistance(BlockDistance::CLOSE);
+
+    stpInfos["waller_2"].setPositionToDefend(field.getOurGoalCenter());
+    stpInfos["waller_2"].setBlockDistance(BlockDistance::CLOSE);
+
+    stpInfos["waller_3"].setPositionToDefend(field.getOurBottomGoalSide());
+    stpInfos["waller_3"].setBlockDistance(BlockDistance::CLOSE);
+}
+
 void DefendShot::calculateInfoForDefenders() noexcept {
-    stpInfos["defender_1"].setPositionToDefend(field.getOurTopGoalSide());
+    auto enemyRobots = world->getWorld()->getThem();
+
+    auto enemyClosestToBall = world->getWorld()->getRobotClosestToBall(world::them);
+
+    enemyRobots.erase(std::remove_if(enemyRobots.begin(), enemyRobots.end(),
+                                     [&](const auto enemyRobot) -> bool { return enemyClosestToBall && enemyRobot->getId() == enemyClosestToBall.value()->getId(); }),
+                      enemyRobots.end());
+
+    auto enemyClosestToOurGoalOne= world->getWorld()->getRobotClosestToPoint(field.getOurGoalCenter(), enemyRobots);
+
+    enemyRobots.erase(std::remove_if(enemyRobots.begin(), enemyRobots.end(),
+                                     [&](const auto enemyRobot) -> bool { return enemyClosestToOurGoalOne && enemyRobot->getId() == enemyClosestToOurGoalOne.value()->getId(); }),
+                      enemyRobots.end());
+
+    auto enemyClosestToOurGoalTwo = world->getWorld()->getRobotClosestToPoint(field.getOurGoalCenter(), enemyRobots);
+
+    enemyRobots.erase(std::remove_if(enemyRobots.begin(), enemyRobots.end(),
+                                     [&](const auto enemyRobot) -> bool { return enemyClosestToOurGoalTwo && enemyRobot->getId() == enemyClosestToOurGoalTwo.value()->getId(); }),
+                      enemyRobots.end());
+
```

Comment:
Instead of using erase and remove_if, you should use erase_if. This is more concise and readable, but also safer, as calling erase on the iterator returned by remove_if can give an error if remove_if does not remove anything, as it will return the past-the-end iterator, which can not be erased.